File: 0350_Intersection of Two Arrays II/IntersectionofTwoArraysII.cs
public class Solution {
    public int[] Intersect (int[] nums1, int[] nums2) {
        Array.Sort (nums1);
        Array.Sort (nums2);

        int p1 = 0;
        int p2 = 0;

        List<int> list = new List<int> ();

        while (p1 < nums1.Count () && p2 < nums2.Count ()) {
            if (nums1[p1] == nums2[p2]) {
                list.Add (nums1[p1]);
                p1++;
                p2++;
            } else if (nums1[p1] > nums2[p2]) {
                p2++;
            } else {
                p1++;
            }
        }

        return list.ToArray ();
    }

}

File: 0350_Intersection of Two Arrays II/IntersectionofTwoArraysII.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        List<Integer> ans = new ArrayList<Integer>();
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        
        for(int i=0;i<nums2.length;i++) {
            if(map.containsKey(nums2[i])) {
                map.put(nums2[i], map.get(nums2[i]) + 1);
            }else{
                map.put(nums2[i],1);
            }
        }
        
        for(int i=0;i<nums1.length;i++) {
            if(map.containsKey(nums1[i])) {
                ans.add(nums1[i]);
                map.put(nums1[i], map.get(nums1[i]) - 1);
                if(map.get(nums1[i]) == 0) map.remove(nums1[i]);
            }
        }
        
        return ans.stream().mapToInt(i->i).toArray();
    }
}